File: API/Controllers/PaymentsController.cs
using Core.Entities;
using Core.Interfaces;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Http;
using Microsoft.AspNetCore.Mvc;

namespace API.Controllers
{
    public class PaymentsController(IPaymentService paymentService, 
        IUnitOfWork unitOfWork) :  BaseApiController
    {
        [Authorize]
        [HttpPost("{cartId}")]
        public async Task<ActionResult<ShoppingCart>> CreateorUpdatePaymentIntent(string cartId)
        {
            var cart = await paymentService.CreateOrUpadatePaymentIntent(cartId);

            if (cart == null) return BadRequest("Problem with your cart");

            return Ok(cart);

        }
        [Authorize]
        [HttpGet("delivery-methods")]
        public async Task<ActionResult<IReadOnlyList<DeliveryMethod>>> GetDeliveryMethods()
        {
            return Ok(await unitOfWork.Repository<DeliveryMethod>().ListAllAsync());

        }
    }
}

File: java-api/api/src/main/java/com/aayuskinet/api/controllers/PaymentsController.java
package com.aayuskinet.api.controllers;

import com.aayuskinet.core.entities.DeliveryMethod;
import com.aayuskinet.core.entities.ShoppingCart;
import com.aayuskinet.core.interfaces.IDeliveryMethodRepository;
import com.aayuskinet.core.interfaces.IPaymentService;
import org.springframework.http.ResponseEntity;
import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Optional;

@RestController
@RequestMapping("/api/payments")
public class PaymentsController extends BaseApiController {

    private final IPaymentService paymentService;
    private final IDeliveryMethodRepository deliveryMethodRepository;

    public PaymentsController(IPaymentService paymentService, IDeliveryMethodRepository deliveryMethodRepository) {
        this.paymentService = paymentService;
        this.deliveryMethodRepository = deliveryMethodRepository;
    }

    @PreAuthorize("isAuthenticated()")
    @PostMapping("/{cartId}")
    public ResponseEntity<ShoppingCart> createOrUpdatePaymentIntent(@PathVariable String cartId) {
        Optional<ShoppingCart> cart = paymentService.createOrUpdatePaymentIntent(cartId);

        return cart.map(ResponseEntity::ok)
                .orElseGet(() -> ResponseEntity.badRequest().body(null)); // Problem with your cart
    }

    @PreAuthorize("isAuthenticated()")
    @GetMapping("delivery-methods")
    public ResponseEntity<List<DeliveryMethod>> getDeliveryMethods() {
        List<DeliveryMethod> deliveryMethods = deliveryMethodRepository.listAll();
        return ResponseEntity.ok(deliveryMethods);
    }
}
